### PR TITLE
test: Write charge input data to delta table for integration testing

### PR DESCRIPTION
**Pull request by JesperGomes**
## Description

Write charge data into relevant delta tables so that we can perform integration tests on wholesale calculations

Also, the test tables' location is specified on creation, because it makes it easier to clean up data when tests are re-run.

## References

* #0000

### CODE REVIEW

File: source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py [hunk 1]
```diff
@@ -66,14 +68,24 @@ def grid_loss_responsible_test_data(
...
+def metering_points_and_time_series_input_data_written_to_delta_tables(
+    spark: SparkSession,
+    test_files_folder_path: str,
+    input_database_created: None,
+    data_lake_path: str
+) -> None:
+    # Metering point periods
+    spark.sql(
+        f"CREATE TABLE IF NOT EXISTS {infra.INPUT_DATABASE_NAME}.{infra.METERING_POINT_PERIODS_TABLE_NAME} USING DELTA \
+            LOCATION '{data_lake_path}/calculation_input/metering_point_periods'"
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py:87]: Is location needed?

File: source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py [hunk 2]
```diff
@@ -85,6 +97,12 @@ def test_data_written_to_delta_tables(
         f"{infra.INPUT_DATABASE_NAME}.{infra.METERING_POINT_PERIODS_TABLE_NAME}"
     )
 
+    # Time series points
+    spark.sql(
+        f"CREATE TABLE IF NOT EXISTS {infra.INPUT_DATABASE_NAME}.{infra.TIME_SERIES_POINTS_TABLE_NAME} USING DELTA \
+            LOCATION '{data_lake_path}/calculation_input/time_series_points'"
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py:103]: Is location needed?

File: source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py [hunk 3]
```diff
@@ -96,12 +114,68 @@ def test_data_written_to_delta_tables(
...
+
+    charge_master_data_periods_df = spark.read.csv(
+        f"{test_files_folder_path}/ChargeMasterDataPeriods.csv",
+        header=True,
+        schema=charge_master_data_periods_schema,
+    )
+
+    charge_master_data_periods_df.write.format("delta").mode("overwrite").option("overwriteSchema", "true").saveAsTable(
+        f"{infra.INPUT_DATABASE_NAME}.{infra.CHARGE_MASTER_DATA_PERIODS_TABLE_NAME}"
+    )
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py:138]: Applies to all similar: How about using a reusable function like this?
```suggestion
    write_test_data_to_table(
        f"{test_files_folder_path}/ChargeMasterDataPeriods.csv",
        f"{infra.INPUT_DATABASE_NAME}.{infra.CHARGE_MASTER_DATA_PERIODS_TABLE_NAME}"
        charge_master_data_periods_schema,
    )
```

File: source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py [hunk 4]
```diff
@@ -96,12 +114,68 @@ def test_data_written_to_delta_tables(
...
+
+
 @pytest.fixture(scope="session")
 def executed_balance_fixing(
     spark: SparkSession,
     calculator_args_balance_fixing: CalculatorArgs,
     migrations_executed: None,
-    test_data_written_to_delta_tables: None,
+    metering_points_and_time_series_input_data_written_to_delta_tables: None,
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py:178]: What do you think about this?
```suggestion
    energy_test_data_written_to_delta: None,
```

File: source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py [hunk 5]
```diff
@@ -119,7 +193,8 @@ def executed_wholesale_fixing(
     spark: SparkSession,
     calculator_args_wholesale_fixing: CalculatorArgs,
     migrations_executed: None,
-    test_data_written_to_delta_tables: None,
+    metering_points_and_time_series_input_data_written_to_delta_tables: None,
+    charge_input_data_written_to_delta_tables: None,
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py:197]: How about something similar to this?
```suggestion
    energy_input_data_written_to_delta: None,
    price_input_data_written_to_delta: None,
```

File: source/databricks/calculation-engine/tests/integration/calculator/test_files/ChargeLinkPeriods.csv [hunk 6]
```diff
@@ -0,0 +1,551 @@
+charge_id,charge_type,charge_owner,metering_point_id,from_date,to_date
+40000,tariff,5790001330552,578710000000000103,2018-01-02 00:00:00,2018-01-03 00:00:00
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/integration/calculator/test_files/ChargeLinkPeriods.csv:2]: What is the time zone? Should these be danish midnight in UTC?

File: source/databricks/calculation-engine/tests/integration/calculator/test_files/old/ChargeLinkPeriods.csv
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/tests/integration/calculator/test_files/old/ChargeLinkPeriods.csv:1]: Any reason to not delete all these old files?

Comment (JesperGomes) [reply source/databricks/calculation-engine/tests/integration/calculator/test_files/old/ChargeLinkPeriods.csv:1]: Oops. Thanks

File: source/databricks/calculation-engine/tests/integration/calculator/test_files/ChargeLinkPeriods.csv [hunk 7]
```diff
@@ -0,0 +1,551 @@
+charge_id,charge_type,charge_owner,metering_point_id,from_date,to_date
```
[the remark this answers is not in the document]
Comment (JesperGomes) [reply source/databricks/calculation-engine/tests/integration/calculator/test_files/ChargeLinkPeriods.csv:1]: maybe - it's not part of the contract at this point. I will add it as a topic for potential changes to the contract.

File: source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py (hunk 1 above)
Comment (JesperGomes) [reply source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py:87]: I had some issues cleaning up test data between test runs, so it makes it easier to control with a specified location. There might be a another solution without location, but I didn't find it important enough to spend more time on it.

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**